resources/src/main/java/org/robolectric/res/StringResources.java

```diff
@@ -96,4 +96,8 @@ static String escape(String text) {
       throw new IllegalArgumentException("Invalid code point: \\u" + codePoint, e);
     }
   }
+
+  private static String removeUnescapedDoubleQuotes(String input) {
+    return input.replaceAll("\"", "");
```

Comment:
I don't think there's a clean/simple regex that can handle all the edge cases. For example:

```
<string name="s1">a\"b</string> <!- quote remains -->
<string name="s2">a\\"b</string> <!- quote removed -->
<string name="s3">a\\\"b</string> <!-- quote remains -->
<string name="s4">a\\\\"b</string> <!-- quote removed -->
```

You'd basically have to write a regex that handles an even number of backslashes that precedes the quote. It's probably doable but it would be messy.

Can you modify `escape` method to handle this? It would have to go char-by-char instead of scanning for backslashes.